package.json

```diff
@@ -42,6 +42,6 @@
     "jest-circus": "^24.9.0",
     "prettier": "^1.19.1",
     "ts-jest": "^24.2.0",
-    "typescript": "^3.7.5"
+    "typescript": "^4.6.3"
```

Comment:
Maybe separate out the major version bump of Typescript as a separate PR and just leave this PR as just the code changes?

Reply:
I could not get the unit tests running with typescript 3. I could back this out, but there would be no way to test my fixes then. I'll give it another go though, see what the issue was

Reply:
Maybe double check which version of Node.js you're using? The repo is set to use 16 LTS when it's used via Github Actions. I just opened #29 to try and ensure the development and CI runs match this. Was able to run it locally with no issues. Awaiting approval from a maintainer to run the CI and see if that holds true as well for it's CI tests.